src/main/java/hudson/plugins/s3/S3BucketPublisher.java

```diff
@@ -141,8 +144,13 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath ws, @Nonnull Launc
             final List<FingerprintRecord> artifacts = Lists.newArrayList();
 
             for (Entry entry : entries) {
+                if (buildAborted) {
```

Comment:
If build is aborted, it's aborted for all entries. So, you can easily move this check on top of this method and remove `buildAborted` variable.